ls1intum/Artemis
# Bugfix/fix markdown editor bold command

### Checklist
~~- [x] Server: I added multiple integration tests (Spring) related to the features~~
~~- [x] Server: I added `@PreAuthorize` and check the course groups for all new REST Calls (security)~~
~~- [x] Server: I implemented the changes with a good performance and prevented too many database calls~~
~~- [x] Server: I documented the Java code using JavaDoc style.~~
~~- [x] Client: I added `authorities` to all new routes and check the course groups for displaying navigation elements (links, buttons)~~

### Motivation and Context
Open issue link: https://jira.ase.in.tum.de/browse/ARTEMIS-278
The markdown '**' used to be added at the beginning and at the end of selected text regardless if selected text end or start with white space


### Description
Added two loops to count the white-spaces at the beginning and at the end of selected text.
Sliced the selected text into white-space part and actual text part then added the markdowns only around actual text leaving white-space outside

### Steps for Testing
1. navigate to course administration
2. edit an existing exercise
3. write text in one of the editors for example in problem statement text editor
4. select text that start or/and ends with white-space
5. click on the"B" (Bold) button
you should see the markdowns only around actual text
6. click preview to see your edited text

### Screenshots
Instructor view:
![image](https://user-images.githubusercontent.com/50635005/66584984-55d9e880-eb86-11e9-9440-7af75e66969f.png)
![image](https://user-images.githubusercontent.com/50635005/66585011-638f6e00-eb86-11e9-9764-e0c474b63ecf.png)
![image](https://user-images.githubusercontent.com/50635005/66585045-75711100-eb86-11e9-8770-4acab30e81ff.png)
Students view:
![image](https://user-images.githubusercontent.com/50635005/66585122-989bc080-eb86-11e9-8fe7-46494fcc8f2c.png)
